### PR TITLE
Insure parse_version does not return tuple (resolves #476)

### LINKED ISSUE #476: Use parse_version

### PR DESCRIPTION
**Pull request by cket**
Resolves #476

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,8 +21,12 @@
 from version import version
 from pkg_resources import parse_version, require, DistributionNotFound
 
+
 def check_provided(distribution, min_version, max_version=None, optional=False):
     min_version = parse_version(min_version)
+    if isinstance(min_version, tuple):
+        raise RuntimeError("Newer version of setuptools required. Update by running "
```
Comment (hannes-ucsc) [inline setup.py]: I think the version that added PEP-440 is 8.0 http://setuptools.readthedocs.io/en/latest/history.html#id223. Could you 1) verify that 8.0 behaves as desired, 2) verify that a version before 8.0 does not behave as desired, i.e. returns a tuple and 3) mention 8.0 as the minimally required version in this error message?

**Comment by cket** [comment]
I did more testing and parse_version doesn't seem to work as expected - we should hold off on margin this

**Comment by cket** [comment]
Never mind, it was an issue with the ranges I was testing with not parse_version itself

**Comment by cket** [comment]
@hannes-ucsc ping for review, I did the requested testing and it worked as expected

File: setup.py [hunk 2]
```diff
@@ -21,8 +21,12 @@
 from version import version
 from pkg_resources import parse_version, require, DistributionNotFound
 
+
 def check_provided(distribution, min_version, max_version=None, optional=False):
     min_version = parse_version(min_version)
+    if isinstance(min_version, tuple):
+        raise RuntimeError("Setuptools version 8.0 or newer required. Update by running "
+                           "'pip install setuptools --upgrade'")
```
Comment (hannes-ucsc) [inline setup.py]: `pip install --upgrade setuptools`

**Pull request merged**